connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -217,35 +221,58 @@ def __get_cosign_validated_digests(self, image: str, trust_root: dict):
             )
         return digests.pop()
 
-    # async def __invoke_cosign(self, image: str, key: str):
-    def __invoke_cosign(self, image: str, key: str):
+    def __validate_using_trust_root(self, image: str, trust_root: TrustRoot):
         """
-        Invoke the Cosign binary in a subprocess for a specific `image` given a `key` and
-        return the returncode, stdout and stderr. Will raise an exception if Cosign times out.
+        Call the `CosignValidator.__invoke_cosign` method, using a specific trust anchor.
+        Depending on the type of trust anchor, the `CosignValidator.__invoke_cosign`
+        method will be called with different arguments.
         """
-        pubkey_config, env_vars, pubkey = CosignValidator.__get_pubkey_config(key)
+        # reminder when implementing RSA validation:
+        # ["--key", "/dev/stdin", self.value.save_pkcs1()]
+
+        # reminder when implementing Keyless validation:
+        # ["--cert-email", self.value, b""]
 
+        if isinstance(trust_root, ECDSAKey):
+            return self.__invoke_cosign(
+                image, ["--key", "/dev/stdin", trust_root.value.to_pem()]
+            )
+        elif isinstance(trust_root, KMSKey):
+            return self.__invoke_cosign(image, ["--key", trust_root.value, b""])
+        msg = (
+            "The trust_root type {tr_type} is unsupported for a validator of type"
+            "{val_type}."
+        )
+        raise WrongKeyError(message=msg, tr_type=type(trust_root), val_type="cosign")
+
+    def __invoke_cosign(self, image: str, tr_args: list):
+        """
+        Invoke the Cosign binary in a subprocess for a specific `image` given a
+        `trust_root` and return the returncode, stdout and stderr.
+
+        Raises an exception if Cosign times out.
+        """
+        option_kword, inline_tr, trust_root = tr_args
```

Comment:
This seems to assume a very specifc config of the tr_args, which is not apparent from the interface. Is __function, so maybe alright, but maybe could be improved

Reply:
tried to make it more clear in the functions description